### ISSUE TITLE
Callback. more than three buttons

### ISSUE THREAD

**Issue by himofftheviz** [JohannBlais/InformationBox#13]
Very good messagebox, but documentation is a bit thin. Can you answer the following please?

Basic constructor ... InfoBox.InformationBox.Show( ... intellisense shows the first parameter signature as "string text, params object[] Parameters ). The description mentions : "The constructor accepts an undefined number of parameters, in an undefined order." How does this basic constructor work? In notice that the enum value for InfoBox.InformationBoxIcon.Information and InfoBox.InformationBoxButtons.YesNo are both 4.

How does the callback feature work - can you provide an example?
Is it possible to have more than three buttons?

Thanks

**Comment by JohannBlais** [comment]
Hi,

**TL;DR**
I recommend that you use named parameters to pass the different configuration parameters to the constructor. If you use the Designer, you will be able to use the code generator to get an example how it works.
No you can't have more than 3 buttons. I made this decision both to keep the code clear (at that time) and because it is not recommended from a UI perspective.

**Long version**
The Box was first created before named parameters were available in the C# language, so I reverted at that time to a `params object[]` parameter-passing style. It enables you to basically pass _anything_ to the constructor. Depending on the types of the `object`, it is interpreted as the corresponding configuration parameter. If you pass a `InformationBoxTitleIconStyle`, it will be mapped to the _Title Icon_ configuration parameter. The mapping table is in the header documentation for the [constructor](https://github.com/JohannBlais/InformationBox/blob/be92489a249531608d02ed0090eded4682661dab/InfoBox/InformationBox.cs#L21)
Parameters that cannot be mapped are ignored.
You can use integer values to the constructor, but you have to cast them as the desired enum type upfront otherwise the constructor will ignore it.
The Callback method is passed as a parameter as well, you can find an example [here](https://github.com/JohannBlais/InformationBox/blob/master/docs/Modeless%20boxes.md)
Please note that most the of the issues described in this section go away if you use named parameters constructor instead of the `params object[]` constructor.

**Comment by himofftheviz** [comment]
Thank you. I see now that Callback is really only for modaless.
I notice most of the things I needed to know was in documentation. Stupid question - where is the link to the documentation from the github informationbox project page?

**Comment by JohannBlais** [comment]
There is apparently no link from the project page, that is something on my list of future improvements :)

**Comment by himofftheviz** [comment]
Haha!
Why not grab the bull by the horns and do it now? Strike while the iron's hot.
I suggest you edit the line  "Find out more about the inner working of InformationBox 'on my blog' ". (where 'on my blog' is a link is your blogspot) and replace the link with "https://github.com/JohannBlais/InformationBox/blob/master/docs".

Regardless, thanks very much for your help: it's a smart little messagebox which I intend to use.

**Comment by JohannBlais** [comment]
Fair enough.

**Comment by JohannBlais** [comment]
You might want to have a look at the latest version introducing the ability to have 3 custom buttons. I know it is not 4, but at least you can give each a specific meaning.

**Comment by cosmic-tiger** [comment]
Sorry for late reply - thanks very much for this, it's great.
